src/umpire/resource/FileMemoryResource.cpp

```diff
@@ -0,0 +1,107 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+//////////////////////////////////////////////////////////////////////////////
+#ifndef UMPIRE_FileMemoryResource_INL
+#define UMPIRE_FileMemoryResource_INL
+
+#include "umpire/resource/FileMemoryResource.hpp"
+
+#include "umpire/util/Macros.hpp"
+
+#include <errno.h>
+#include <string.h>
+#include <fcntl.h>
+#include <ctime>
+#include <sys/mman.h>
+#include <unistd.h>
+#include <utility>
+#include <stdlib.h>
+#include <sys/mman.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+
+namespace umpire {
+namespace resource {
+
+int FILE_COUNTER = 0;
```

Comment:
`FILE_COUNTER` should be placed inside the `FileMemoryResource` class as a private static member variable.